CryptomatorFileProvider/DB/DatabaseHelper.swift

```diff
@@ -10,17 +10,23 @@ import CryptomatorCloudAccessCore
 import FileProvider
 import Foundation
 import GRDB
+public protocol DatabaseHelping {
+	func getDatabaseURL(for domain: NSFileProviderDomain) -> URL
+	func getMigratedDB(at databaseURL: URL, fileCoordinator: NSFileCoordinator) throws -> DatabaseWriter
+}
+
+public struct DatabaseHelper: DatabaseHelping {
+	public static let `default` = DatabaseHelper()
 
-public enum DatabaseHelper {
-	public static func getDatabaseURL(for domain: NSFileProviderDomain) -> URL {
+	public func getDatabaseURL(for domain: NSFileProviderDomain) -> URL {
 		let documentStorageURL = NSFileProviderManager.default.documentStorageURL
 		let domainRootURL = documentStorageURL.appendingPathComponent(domain.pathRelativeToDocumentStorage)
 		return domainRootURL.appendingPathComponent("db.sqlite")
 	}
 
-	public static func getMigratedDB(at databaseURL: URL) throws -> DatabasePool {
-		let dbPool = try openSharedDatabase(at: databaseURL)
-		try migrate(dbPool)
+	public func getMigratedDB(at databaseURL: URL, fileCoordinator: NSFileCoordinator) throws -> DatabaseWriter {
```

Comment:
Would it make sense to pass the param `providerIdentifier` instead of `fileCoordinator` to deduplicate code? Or is there some benefit that the `fileCoordinator` can be re-used from the caller?

Reply:
No you're right. Resolved by https://github.com/cryptomator/ios/pull/274/commits/47aeb625327a3db6985bc1bda60e00c748d40164.